contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/PropertyCriterion.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.contingency.contingency.list.criterion;
+
+import com.google.common.collect.ImmutableList;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.IdentifiableType;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ */
+public class PropertyCriterion implements Criterion {
```

Comment:
As identified by gridsuite project, it's too restrictive to check properties only on the identifiable itself:
for example, we would like to be able to check the property on the substation containing the identifiable.

In order to achieve this, we could add an enum to that criterion with values "SELF / VOLTAGE_LEVEL / SUBSTATION".
Then at filter time, we will get the property either from the identifiable itself, or from the voltage level, or from the substation.

It also means that for multi-sided elements, we will need to allow multiple property criterions ... (2 for branches, 3 for 3WT)